### PR TITLE
commit changelog,localsetup,readme

### PR DESCRIPTION
**Pull request by devdatta-egov**
added readme, changelog and localesetup files

## Summary by CodeRabbit

- **New Features**
  - Introduced Plan Service module version 1.0.0 with functionalities for plan validation, creation, update, and search.

- **Documentation**
  - Updated `README.md` with comprehensive service overview, key features, API documentation, and setup guidelines.
  - Added detailed local setup instructions in `LOCALSETUP.md`.

- **Bug Fixes**
  - Improved null-safety in plan validation logic for checking empty plan configuration IDs.
  - Enhanced query builder to include order-by clause in plan configuration searches.

### CODE REVIEW

File: health-services/plan-service/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Changelog
+All notable changes to this module will be documented in this file.
+
+## 1.0.0 - 2024-06-24
+#### Plan Service
```
[the remark this answers is not in the document]
Comment (devdatta-egov) [reply health-services/plan-service/CHANGELOG.md:5]: Its title and required

File: health-services/plan-service/CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# Changelog
+All notable changes to this module will be documented in this file.
+
+## 1.0.0 - 2024-06-24
+#### Plan Service 
+  1. Plan Service manages: validation of plans, plan search, plan create, plan update.
+  2. Validation of  plan: Plan service validates plan request before it take action on it like update or create.
```
[the remark this answers is not in the document]
Comment (devdatta-egov) [reply health-services/plan-service/CHANGELOG.md:7]: change done

File: health-services/plan-service/LOCALSETUP.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+	[Kafka for windows](https://dzone.com/articles/running-apache-kafka-on-windows-os) or [Kafka for Linux](https://tecadmin.net/install-apache-kafka-ubuntu/)
+5. Install Postman - To install Postman, follow the following links -
+	[Postman for windows](https://www.postman.com/downloads/)
+6. Install Kubectl - Kubectl is the tool that we use to interact with services deployed on our sandbox environment -
+	[kubectl for windows](https://core.digit.org/guides/operations-guide/working-with-kubernetes/installation-of-kubectl)
+    [kubectl for Linux](https://kubernetes.io/docs/tasks/tools/install-kubectl-linux/)
+7. Install aws-iam-authenticator - [if the DIGIT development environment is in AWS](https://docs.aws.amazon.com/eks/latest/userguide/install-aws-iam-authenticator.html)
+8. Install PostgreSQL v14 locally.
+9. Also update DB config values as per your local system config.
+10. Update all dependency service host either on any unified-env or port-forward.
```
[the remark this answers is not in the document]
Comment (devdatta-egov) [reply health-services/plan-service/LOCALSETUP.md:36]: DOne

File: health-services/plan-service/README.md [hunk 4]
```diff
@@ -1,43 +1,58 @@
...
-- plan-config-update-topic
+## Environment Variables
 
-- save-plan
-- update-plan
+- Configure environment variables such as `db-host`, `db-name`, `db-url`, `domain`, and other DIGIT core platform services configurations before deployment. Example configurations can be found in the [unified-qa.yaml](https://github.com/egovernments/DIGIT-DevOps/blob/unified-env/deploy-as-code/helm/environments/unified-qa.yaml).
+
+## MDMS Configuration
+
+- Configure MDMS data for Plan Service including UOM config, Metric config, Assumptions config, Input rules, Output rules, Campaign Based Schema, Microplan status, Map layers, Map filters, Preview Aggregates, and UI configs. Reference data can be found [here](https://github.com/egovernments/egov-mdms-data/tree/UNIFIED-QA/data/mz/health/hcm-microplanning).
```
[the remark this answers is not in the document]
Comment (devdatta-egov) [reply health-services/plan-service/README.md:52]: NA

File: health-services/plan-service/LOCALSETUP.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+11. Run spring boot main class
+
+> Note: After running the above, if a Kafka error occurs, ensure that Kafka and Zookeeper are running in the background. If a connection error with another microservice occurs, ensure that the URL mentioned in the external mapping of the data config is correct, or you can port-forward that particular service.
```
[the remark this answers is not in the document]
Comment (devdatta-egov) [reply health-services/plan-service/LOCALSETUP.md:39]: Done

File: health-services/plan-service/CHANGELOG.md [hunk 6]
```diff
@@ -0,0 +1,10 @@
+# Changelog
+All notable changes to this module will be documented in this file.
+
+## 1.0.0 - 2024-06-24
+#### Plan Service 
+  1. Plan Service manages: validation of plans, plan search, plan create, plan update.
+  2. Validation of  plan: Plan service validates plan request before it take action on it like update or create.  
+  3. Plan Create: Plan service creates a plan after successful validation is done. It sends create request on topic to create plan. 
+  4. Plan Update : Plan service creates a plan after successful validation is done. It sends update request on topic to resource estimation service to further process.
+  5. Plan Search: This enables to search plan based on provided search string.
```
[the remark this answers is not in the document]
Comment (devdatta-egov) [reply health-services/plan-service/CHANGELOG.md:10]: Done

File: health-services/plan-service/CHANGELOG.md [hunk 7]
```diff
@@ -0,0 +1,10 @@
+# Changelog
+All notable changes to this module will be documented in this file.
+
+## 1.0.0 - 2024-06-24
+#### Plan Service 
+  1. Plan Service manages: validation of plans, plan search, plan create, plan update.
+  2. Validation of  plan: Plan service validates plan request before it takes action on it like update or create.  
+  3. Plan Create: Plan service creates a plan after successful validation is done. It sends create request on topic to create plan. 
+  4. Plan Update : Plan service creates a plan after successful validation is done. It sends update request on topic to resource estimation service to further process.
```
[the remark this answers is not in the document]
Comment (devdatta-egov) [reply health-services/plan-service/CHANGELOG.md:9]: NA

File: health-services/plan-service/LOCALSETUP.md [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+- [ ] Redis
+- [ ] Elasticsearch
+- [x] Kafka
+  - [x] Consumer
+  - [x] Producer
+
+
+## Running Locally
+
+### Local setup
```
[the remark this answers is not in the document]
Comment (devdatta-egov) [reply health-services/plan-service/LOCALSETUP.md:17]: NA

File: health-services/plan-service/LOCALSETUP.md [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+3. Install maven locally and configure environment variables.
+4. Install Kafka (version 3.2.0 which is the latest version) - To install and run Kafka locally, follow the following links -
+	[Kafka for windows](https://dzone.com/articles/running-apache-kafka-on-windows-os) or [Kafka for Linux](https://tecadmin.net/install-apache-kafka-ubuntu/)
+5. Install Postman - To install Postman, follow the following links -
+	[Postman for windows](https://www.postman.com/downloads/)
+6. Install Kubectl - Kubectl is the tool that we use to interact with services deployed on our sandbox environment -
+	[kubectl for windows](https://core.digit.org/guides/operations-guide/working-with-kubernetes/installation-of-kubectl)
+    [kubectl for Linux](https://kubernetes.io/docs/tasks/tools/install-kubectl-linux/)
+7. Install aws-iam-authenticator - [if the DIGIT development environment is in AWS](https://docs.aws.amazon.com/eks/latest/userguide/install-aws-iam-authenticator.html)
+8. Install PostgreSQL v14 locally.
```
[the remark this answers is not in the document]
Comment (devdatta-egov) [reply health-services/plan-service/LOCALSETUP.md:34]: NA

File: health-services/plan-service/CHANGELOG.md [hunk 10]
```diff
@@ -0,0 +1,10 @@
+# Changelog
+All notable changes to this module will be documented in this file.
+
+## 1.0.0 - 2024-06-24
+#### Plan Service 
+  1. Plan Service manages: validation of plans, plan search, plan create, plan update.
+  2. Validation of  plan: Plan service validates plan request before it takes action on it like update or create.
```
[the remark this answers is not in the document]
Comment (devdatta-egov) [reply health-services/plan-service/CHANGELOG.md:7]: NA

**Review by shashwat-egov — APPROVED** [review]

**Review by kavi-egov — APPROVED** [review]

**Pull request merged**